scripts/roombot/bot_patrol.gd

```diff
@@ -4,7 +4,7 @@ class_name BotPatrol extends Patrol
 
 
 func process_frame(delta: float) -> EnemyState:
-	if parent.player_detection.is_colliding():
+	if parent.player_detection.get_collider() is Player:
```

Comment:
any specific reason you changed to check if the body is a player? Before, the raycast only interacted with a player object by setting its mask to the players layer, so it functionally did the same thing.

Reply:
Right now, the Raycast ignores walls so the Roombot sees through them.
The Raycast needs to also collide with walls and then check if the collider was Player specifically

Reply:
makes sense :+1: